### PR TITLE
[type checking test files] Boot -> MockBoot

### PR DESCRIPTION
**Pull request by ds300**
### See #1166 for background and approach

In this exciting first edition, we:

- Replaced a bunch of type-unsafe usages of the `Boot` component with a more safe `MockBoot` component which has a smaller API designed specifically for testing, that can grow over time.
- Type-checked some mock relay data.
- Deleted some unused things.

🕺

### CODE REVIEW

**Comment by alloy** [comment]
As mentioned previously, I think we should get rid of all manual data insertion into these containers, but for the time being (and whenever really necessary), you may want to try this approach that somebody shared https://github.com/relay-tools/relay-compiler-language-typescript/issues/29#issuecomment-417101505

File: src/Styleguide/Components/__tests__/ArtistCard.test.tsx [hunk 1]
```diff
@@ -18,39 +19,39 @@ describe("ArtistCard", () => {
...
 
   beforeAll(() => {
-    window.matchMedia = undefined // Immediately set matching media query in Boot
+    window.matchMedia = undefined // Immediately set matching media query in MockBoot
   })
 
   it("is responsive", () => {
     const small = mount(
-      <Boot initialMatchingMediaQueries={["xs"]}>
+      <MockBoot breakpoint="xs">
```
Comment (damassi) [inline src/Styleguide/Components/__tests__/ArtistCard.test.tsx]: `breakpoint` is a bit easier on the hands. At some point if we wanted to make it more correct I think we could add a `mediaQuery` prop and merge the two together as it's passed into Boot.

File: src/Styleguide/Elements/__tests__/Select.test.tsx [hunk 2]
```diff
@@ -11,16 +11,16 @@ describe("Select", () => {
 
   it("is responsive", () => {
     const small = mount(
-      <Boot initialMatchingMediaQueries={["xs"]}>
+      <MockBoot breakpoint={"xs"}>
```
Comment (damassi) [inline src/Styleguide/Elements/__tests__/Select.test.tsx]: #superminor - For consistency with the above changes, maybe just `breakpoint='xs`?

File: src/Utils/MockBoot.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Breakpoint } from "@artsy/palette/dist/Theme"
```
Comment (damassi) [inline src/Utils/MockBoot.tsx]: Can import this from the root as all exports are now available there. 

`import { Breakpoint } from '@artsy/palette'`

File: src/Utils/MockBoot.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      context={null as any}
+      user={null as any}
+      relayEnvironment={null as any}
+      resolver={null as any}
+      routes={null as any}
+    >
+      {children}
+    </Boot>
+  )
+}
```
Comment (damassi) [inline src/Utils/MockBoot.tsx]: I dig it 😄 We can expand as needed with time but i like how for now its very explicitly constrained.

**Comment by ds300** [comment]
Thanks for your feedback! I've addressed the issues.

Merge on green

**Pull request merged**